File: backend/models/Appointment.js
const mongoose = require("mongoose");

const appointmentSchema = new mongoose.Schema({
  user: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
  consultant: { type: mongoose.Schema.Types.ObjectId, ref: "Consultant", required: true },

  date: String,
  startTime: String,
  endTime: String,

  status: {
    type: String,
    enum: ["PENDING", "CONFIRMED", "CANCELLED", "COMPLETED"],
    default: "PENDING",
  },

  otp: String,
  otpExpiresAt: Date,
});

module.exports = mongoose.model("Appointment", appointmentSchema);

File: appointment_app/lib/models/appointment.dart
class Appointment {
  final String id;
  final String date;
  final String startTime;
  final String endTime;
  final String status;
  final Map<String, dynamic>? consultant;

  Appointment({
    required this.id,
    required this.date,
    required this.startTime,
    required this.endTime,
    required this.status,
    this.consultant,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['_id'] ?? '',
      date: json['date'] ?? '',
      startTime: json['startTime'] ?? '',
      endTime: json['endTime'] ?? '',
      status: json['status'] ?? '',
      consultant: json['consultant'],
    );
  }
}